### PR TITLE
fix: LSDV-4600: Lead time for TextArea

### PR DESCRIPTION
**Pull request by juliosgarbi**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)

- [ ] Product design
- [x] Frontend

### Describe the reason for change
Annotation results now have a "lead_time"value in export data for textareas. The value reflects how much time the annotator spent typing values in this TextArea. Lead Time  is calculated with minimal interaction time 0.5s and with debounce of the same 0.5s, for example:
- slow typer, one letter per second, 10 letters: 10*0.5s = 5s lead time
- fast typer, all 10 letters in 2s: 2+0.5s debounce = 2.5s
 
#### What does this fix?
This solves a time tracking problem so that typing time of annotators can be calculated. 

#### What libraries were added/updated?
N/A

#### Does this change affect performance?
no

#### Does this change affect security?
no

#### What alternative approaches were there?
N/A

#### What feature flags were used to cover this change?
`fflag_fix_front_lsdv_4600_lead_time_27072023_short`

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [x] e2e
- [x] integration
- [ ] unit

### Which logical domain(s) does this change affect?
Textarea, Lead Time

### CODE REVIEW

File: src/mixins/LeadTime.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { types } from 'mobx-state-tree';
+import debounce from  'lodash.debounce';
+
+const DEBOUNCE_INTERVAL = 500;
```
Comment (bmartel) [inline src/mixins/LeadTime.js:4]: Does this have the chance to miss the debounced setting of leadTime, where it doesn't get updated in time for draft save?

Comment (bmartel) [reply src/mixins/LeadTime.js:4]: I was also curious as to why we are debouncing it by 500ms?

Comment (bmartel) [reply src/mixins/LeadTime.js:4]: resolving after discussion with @juliosgarbi, this is meant to capture leadtime with a minimum interaction time, so repetitive typing is captured in a single window of time, and one-off typing gets considered as separate windows of 0.5 seconds in the leadtime calculation.

**Review by bmartel — APPROVED** [review]

File: src/mixins/LeadTime.js (hunk 1 above)
Comment (hlomzik) [reply src/mixins/LeadTime.js:4]: Opening this back, as it's worth to add a comment about this var and it's purpose. Like "Minimal interaction time to count as well as debounce for multiple consequential interactions, so they can be counted as one long interaction window."

**Review by hlomzik — CHANGES_REQUESTED** [review]
It's all good, but little comments+fix would be nice, and the main thing — Lead Time should be described in a PR and in mixin, as this is the first ever appearance of this calculations in the code.

File: src/mixins/LeadTime.js (hunk 1 above)
Comment (Travis1282) [reply src/mixins/LeadTime.js:4]: resolving as there is no change requested

Comment (hlomzik) [reply src/mixins/LeadTime.js:4]: em... that was an action requested, to add comment. we have a very few comments in a code and we should go toward more comments and cleaner and readable code.

File: src/mixins/LeadTime.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+const LeadTimeMixin = types
+  .model({
+    leadTime: 0,
+  })
+  .volatile(() => ({
+    startTime: 0,
+  }))
+  .actions(self => ({
+    internalCountTime(callTime) {
+      const now = +new Date();
```
[the remark this answers is not in the document]
Comment (hlomzik) [reply src/mixins/LeadTime.js]: 2 more occurencies left

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/mixins/LeadTime.js (hunk 1 above)
Comment (Travis1282) [reply src/mixins/LeadTime.js]: My apologies I misunderstood what you meant by comment. There are lots of comments in the file already would you settle for a more descriptive debounce variable name?

File: src/mixins/LeadTime.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Travis1282) [reply src/mixins/LeadTime.js]: ok!

File: src/tags/control/TextArea/TextArea.js [hunk 3]
```diff
@@ -306,6 +320,33 @@ const Model = types.model({
       }
     },
 
+    toStateJSON() {
```
[the remark this answers is not in the document]
Comment (Travis1282) [reply src/tags/control/TextArea/TextArea.js:347]: removed

File: src/mixins/LeadTime.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const LeadTimeMixin = types
+  .model({
+    leadTime: 0,
+  })
+  .volatile(() => ({
+    startTime: 0,
+  }))
+  .actions(self => ({
+    internalCountTime(callTime) {
+      const now = +Date.now();
```
Comment (nicholasrq) [inline src/mixins/LeadTime.js:18]: `Date.now()` returns a number, you don't need to convert it to number

**Review by nicholasrq — APPROVED** [review]

File: src/mixins/LeadTime.js (hunk 4 above)
Comment (hlomzik) [reply src/mixins/LeadTime.js:18]:
```suggestion
      const now = Date.now();
```

**Review by Gondragos — APPROVED** [review]

**Pull request merged**